gui/console.cpp

```diff
@@ -646,8 +660,18 @@ void ConsoleDialog::historyScroll(int direction) {
 		_history[_historyIndex] = getUserInput();
 
 	// Advance to the next line in the history
+	// Note that due to temporarily storing the user input line in the _history table,
+	//      without executing an AddToHistory() call,
+	//      that user input line is stored into the slot _historyIndex
+	//      where the next committed command will replace it.
+	//      However, since this slot is still a slot from the _history table,
+	//      when the table is full (kHistorySize entries) and while scrolling the history upwards
+	//      the user can reach this slot (top most) and get their user input again instead of a historic entry.
+	//      We prevent this by stopping upwards navigation one slot earlier, when the table is full.
```

Comment:
The wrapping (inconsistent line lengths) and indentation for this comment block seems strange to me. But that's again nitpicking.

Reply:
Improved the wrapping in the recent commit. The indentation was because I wanted to emphasize that this was all under the "NOTE" entry (I capitalized that too in the recent commit). 
https://github.com/scummvm/scummvm/pull/3804/commits/f58aa17c4eead86f0116be320676aff5f5fc9c90